src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -72,7 +72,7 @@ createFixRatePaymentClaims dateToTime schedule periodicSchedule useAdjustedDates
 -- | Calculate a floating rate amount for each payment date and create claims.
 -- This is a general function that supports both LIBOR and SOFR-COMPOUND reference rates.
 createFloatingRatePaymentClaims : (Date -> Time) -> Schedule -> PeriodicSchedule -> Bool ->
-  Decimal -> Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> FloatingRate ->
+  Decimal -> Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> Optional FloatingRate ->
```

Comment:
What happens if I set `None`?

Reply:
Then it only uses the fix `couponRate`, which is otherwise used as a floating coupon spread. In other words, this function can generate claims for both fix and floating rates. I should probably rename it to `createRatePaymentClaims`.